src/elements/content-sidebar/versions/VersionsItem.js

```diff
@@ -86,9 +98,12 @@ const VersionsItem = ({
     const isDownloadable = !!is_download_available;
     const isLimited = versionCount - versionInteger >= versionLimit;
     const isRestricted = isWatermarked && !isCurrent; // Watermarked files do not support prior version preview
+    const isRetentionActive =
+        !!versionAppliedTime && (!versionDispositionTimestamp || versionDispositionTimestamp > new Date().getTime());
 
     // Version action helpers
     const canPreview = can_preview && !isDeleted && !isLimited && !isRestricted;
+    const enableDelete = !isRetentionActive;
```

Comment:
Looking at this again, is this variable needed? It's used only once for the action component.